docs/BuildOnLinuxOSX.md

```diff
@@ -97,6 +99,22 @@ make: *** [Makefile:146: all] Error 2.
 
 The suggested workaround until jsoniter is fixed is as follows: install maven (e.g. `brew install maven`) and run `alias nproc="sysctl -n hw.logicalcpu"` in your shell.
 
+#### Protobuf issue (Mac M1, specific to protobuf 3.20.3 which is currently required)
+
+On Mac M1, you may have some issues building protobuf. In particular, you may fail to install onnx (via `pip install -e third_party/onnx`) or you may fail to compile `onnx-mlir` (no arm64 symbol for `InternalMetadata::~InternalMetadata`).
+
+The first failure is likely an issue with having multiple versions of protobuf.
+Installing a version with `brew` was not helpful (version 3.20.3 because of a known bug that can be corrected with a patch below).
+Uninstall the brew version, and make sure you install the right one with pip: `pip install protobuf==3.20.3`.
```

Comment:
is it counterproductive to mention how to install 3.20.3 if it doesn't work to compile onnx-mlir and the real recommendation is to build from source?

Reply:
Surprisingly, I had to have it installed via pip because otherwise `pip install` would automatically download via pip the latest (3.4.*), which is incompatible. So it needs to have 3.20.3 installed via pip to make the `pip install -e thrid_party/onnx` install happy, but having the code version on the path for the building of onyx-mlir compiler.

Would you like me to be more explicit about this point?

Reply:
I see, now I see your description makes good sense

however, I think it's actually simpler to just "pip install onnx" (or "pip instlal onnx==$(cat third_party/onnx/VERSION_NUMBER)" if pip doesn't pick the right version automatically), then you don't need pip's protobuf

Reply:
I have not tried this. For some reason, I assumed that we might need the third_party implementation... but if that is not the case, then it's even easier.

I may have to uninstall protobuf to try it out. Is that all you did? Namely no installation of protobuf, just `pip install onnnx=3.20.3` ?

Reply:
If so, I can write it as an alternative to try. Our CIs definitely don't go that route.

Reply:
> Is that all you did? Namely no installation of protobuf, just `pip install onnnx=3.20.3` ?

it's the onnx version (1.14.0), not protobuf version

I got the python onnx module with `pip install onnx` (it happened to pick the correct version, I think pip install onnx==1.14.0` would also have worked)